enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,441 @@
+know the type of the cluster.
+
+On a fresh install, CVO would detect the architecture from the metadata JSON inside the individual architecture's payload inside the manifestlisted
+release payload. The `release.openshift.io/architecture` property would be set to "multi" denoting a heterogeneous payload. It would then retrieve the
+heterogeneous graph to service upgrade requests and set the ClusterVersion status architecture field.
+
+The migration [support][OTA-653] is planned for phase 3 and users can use the explicit `--to-image` flag along with `oc adm upgrade` to convert the
+cluster to a heterogeneous cluster in the phase 2 timeframe.
+
+Cincinnati needs to [support][OTA-580] parsing manifestlist images to be able to upgrade to a heterogeneous architecture payload. There will be a
+new architecture option added to it called "multi".
+
+### Imagestreams
+
+Imagestreams need to [better support][IR-192] manifestlist images. As of today, the imagestreams support only import of the image matching the control plane
+architecture. The image registry which is based on the docker distribution supports manifestlisted images, but the Imagestreams APIs around it
+do not. In order to fully support importing and local pullthrough of manifestlisted images,the imagestreams APIs would have to change to support
+manifestlists. This would include changes to:
+- `ImageStreamLayers` API which is needed to authorize pulls from the registry
+- `Image` API which would need to also contain all the child manifests in the manifestlist
+- `ImageStreamImport` API - the `ImageStreamImportStatus` would have to record the child manifests that were imported
+- `ImageStreamImage` API changes to support returning information about the child manifests.
+
+The pruner would also need some careful changes to make sure that it does not evict the manifestlisted image if any of the child manifests are in
+use. Conversely, all child manifests are also removed when an imagestream tag referencing a manifestlisted image is removed.
+
+oc tooling around imagestream commands would also need to change and factor in manifestlists in case of commands like `oc describe` which would also
+be enhanced to present the manifestlist information.
+
+This manifestlist support [enhancement][manifestlist-support] which was filed a while ago explains in detail the API changes needed and the different
+parts of the imagestreams and imageregistry functionality which would need to be changed to support manifestlisted images.
+
+### Samples
+
+As part of the changes to the Imagestreams API, the ImagestreamImport API's spec field would include a [new flag][IR-214] which would import the manifestlist
+rather than importing only an image from the manifestlist. The samples imagestreams would be updated to set this new flag. Not all samples will support
+all architectures, so in the future imagestreams would have to be annotated to specify which architectures they support.
+
+### OLM
+
```

Comment:
@joelanford any concerns over the OLM section. if not, could you provide a verbal approval for this section?